patches/api/0203-Spawn-Reason-API.patch

```diff
@@ -22,33 +22,130 @@ index a89fff5c164881be0286ec2240e94dd5883ecc40..e55f6e2baf35dbd91c433ab9e62713ea
++    default <T extends Entity> T spawn(@NotNull Location location, @NotNull Class<T> clazz, @NotNull CreatureSpawnEvent.SpawnReason reason) throws IllegalArgumentException {
 +        return spawn(location, clazz, reason, null);
 +    }
 +
 +    @NotNull
-+    public default <T extends Entity> T spawn(@NotNull Location location, @NotNull Class<T> clazz, @NotNull CreatureSpawnEvent.SpawnReason reason, @Nullable Consumer<T> function) throws IllegalArgumentException {
++    default <T extends Entity> T spawn(@NotNull Location location, @NotNull Class<T> clazz, @NotNull CreatureSpawnEvent.SpawnReason reason, @Nullable Consumer<T> function) throws IllegalArgumentException {
 +        return spawn(location, clazz, function, reason);
 +    }
 +
 +    @NotNull
-+    public default Entity spawnEntity(@NotNull Location loc, @NotNull org.bukkit.entity.EntityType type, @NotNull CreatureSpawnEvent.SpawnReason reason) {
++    default Entity spawnEntity(@NotNull Location loc, @NotNull org.bukkit.entity.EntityType type, @NotNull CreatureSpawnEvent.SpawnReason reason) {
 +        return spawn(loc, (Class<Entity>) type.getEntityClass(), reason, null);
 +    }
 +
 +    @NotNull
-+    public default Entity spawnEntity(@NotNull Location loc, @NotNull org.bukkit.entity.EntityType type, @NotNull CreatureSpawnEvent.SpawnReason reason, @Nullable Consumer<Entity> function) {
++    default Entity spawnEntity(@NotNull Location loc, @NotNull org.bukkit.entity.EntityType type, @NotNull CreatureSpawnEvent.SpawnReason reason, @Nullable Consumer<Entity> function) {
 +        return spawn(loc, (Class<Entity>) type.getEntityClass(), reason, function);
 +    }
 +
 +    @NotNull
-+    public <T extends Entity> T spawn(@NotNull Location location, @NotNull Class<T> clazz, @Nullable Consumer<T> function, @NotNull CreatureSpawnEvent.SpawnReason reason) throws IllegalArgumentException;
++    <T extends Entity> T spawn(@NotNull Location location, @NotNull Class<T> clazz, @Nullable Consumer<T> function, @NotNull CreatureSpawnEvent.SpawnReason reason) throws IllegalArgumentException;
 +    // Paper end
  
      /**
       * Creates a new entity at the given {@link Location} with the supplied
+diff --git a/src/main/java/org/bukkit/World.java b/src/main/java/org/bukkit/World.java
+index 352a20da402b3b0182b7a0c69d94397f42054053..6a9604404ed07843c9da91a929651c0d1f1e9d6d 100644
+--- a/src/main/java/org/bukkit/World.java
++++ b/src/main/java/org/bukkit/World.java
+@@ -706,7 +706,27 @@ public interface World extends RegionAccessor, WorldInfo, PluginMessageRecipient
+      * @return Arrow entity spawned as a result of this method
+      */
+     @NotNull
+-    public <T extends AbstractArrow> T spawnArrow(@NotNull Location location, @NotNull Vector direction, float speed, float spread, @NotNull Class<T> clazz);
++    // Paper start
++    default  <T extends AbstractArrow> T spawnArrow(@NotNull Location location, @NotNull Vector direction, float speed, float spread, @NotNull Class<T> clazz) {
```

Comment:
*absolute nitpick* but there are two spaces between `default` and `<T extends AbstractArrow>` :sweat_smile:

Reply:
..holy, you should really see that comment: https://github.com/PaperMC/Paper/commit/c6d737528ddb14e360471b39eb97f65e959f88a2#diff-31aba21b5bf05c168ca05351018164e05fc3b26f84ce9d4ac2e19c0c0bbdbf33R22 :eyes:  

But i will fix that, thanks :heart:.

Reply:
Yea I already left my comment on the merged commit in regards to that PR. Really just a) if you are gonna work here anyway, you can just fix it XD